test/CMakeLists.txt

```diff
@@ -1,312 +1,144 @@
+  endforeach()
+endfunction(make_test)
+
+make_test(approx1.cpp)
+make_test(approx2.cpp)
+make_test(array.cpp)
+make_test(assign.cpp)
+make_test(backend.cpp)
+make_test(basic.cpp)
+make_test(bilateral.cpp)
+make_test(binary.cpp)
+make_test(blas.cpp)
+make_test(canny.cpp)
+make_test(cast.cpp)
+make_test(cholesky_dense.cpp)
+make_test(clamp.cpp)
+make_test(compare.cpp)
+make_test(complex.cpp)
+make_test(constant.cpp)
+make_test(convolve.cpp)
+make_test(corrcoef.cpp)
+make_test(covariance.cpp)
+make_test(diagonal.cpp)
+make_test(diff1.cpp)
+make_test(diff2.cpp)
+make_test(dog.cpp)
+make_test(dot.cpp)
+make_test(empty.cpp)
+make_test(fast.cpp)
+make_test(fft.cpp)
+make_test(fft_large.cpp)
+make_test(fft_real.cpp)
+make_test(fftconvolve.cpp)
+make_test(flat.cpp)
+make_test(flip.cpp)
+make_test(gaussiankernel.cpp)
+make_test(gen_assign.cpp)
+make_test(gen_index.cpp)
+make_test(getting_started.cpp)
+make_test(gfor.cpp)
+make_test(gloh_nonfree.cpp)
+make_test(gradient.cpp)
+make_test(gray_rgb.cpp)
+make_test(hamming.cpp)
+make_test(harris.cpp)
+make_test(histogram.cpp)
+make_test(homography.cpp)
+make_test(hsv_rgb.cpp)
+make_test(iir.cpp)
+make_test(imageio.cpp)
+make_test(index.cpp)
+make_test(info.cpp)
+make_test(internal.cpp)
+make_test(inverse_dense.cpp)
+make_test(iota.cpp)
+make_test(ireduce.cpp)
+make_test(jit.cpp)
+make_test(join.cpp)
+make_test(lu_dense.cpp)
+make_test(main.cpp)
+make_test(manual_memory_test.cpp)
+make_test(match_template.cpp)
+make_test(math.cpp)
+make_test(matrix_manipulation.cpp)
+make_test(mean.cpp)
+make_test(meanshift.cpp)
+make_test(medfilt.cpp)
+make_test(median.cpp)
+make_test(memory.cpp)
+make_test(memory_lock.cpp)
+make_test(missing.cpp)
+make_test(moddims.cpp)
+make_test(moments.cpp)
+make_test(morph.cpp)
+make_test(nearest_neighbour.cpp)
+make_test(ocl_ext_context.cpp)
+make_test(orb.cpp)
+make_test(qr_dense.cpp)
+make_test(random.cpp)
+make_test(range.cpp)
+make_test(rank_dense.cpp)
+make_test(reduce.cpp)
+make_test(regions.cpp)
+make_test(reorder.cpp)
+make_test(replace.cpp)
+make_test(resize.cpp)
+make_test(rotate.cpp)
+make_test(rotate_linear.cpp)
+make_test(sat.cpp)
+make_test(scan.cpp)
+make_test(scan_by_key.cpp)
+make_test(select.cpp)
+make_test(set.cpp)
+make_test(shift.cpp)
+make_test(sift_nonfree.cpp)
+make_test(sobel.cpp)
+make_test(solve_dense.cpp)
+make_test(sort.cpp)
+make_test(sort_by_key.cpp)
+make_test(sort_index.cpp)
+make_test(sparse.cpp)
+make_test(sparse_arith.cpp)
+make_test(sparse_convert.cpp)
+make_test(stdev.cpp)
+make_test(susan.cpp)
+make_test(svd_dense.cpp)
+make_test(threading.cpp)
+make_test(tile.cpp)
+make_test(transform.cpp)
+make_test(transform_coordinates.cpp)
+make_test(translate.cpp)
+make_test(transpose.cpp)
+make_test(transpose_inplace.cpp)
+make_test(triangle.cpp)
+make_test(unwrap.cpp)
+make_test(var.cpp)
+make_test(where.cpp)
+make_test(wrap.cpp)
+make_test(write.cpp)
+make_test(ycbcr_rgb.cpp)
```

Comment:
While we are here, can you make the test targets be `test_name_backend`. In the existing setup the test targets are `name_backend` while the example targets are `example_name_backend`